sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -29,17 +29,20 @@ import org.apache.spark.sql.catalyst.util._
 class CSVOptions(
     @transient val parameters: CaseInsensitiveMap[String],
     defaultTimeZoneId: String,
-    defaultColumnNameOfCorruptRecord: String)
+    defaultColumnNameOfCorruptRecord: String,
+    val columnPruning: Boolean)
   extends Logging with Serializable {
 
   def this(
     parameters: Map[String, String],
     defaultTimeZoneId: String,
-    defaultColumnNameOfCorruptRecord: String = "") = {
+    defaultColumnNameOfCorruptRecord: String = "",
+    columnPruning: Boolean = false) = {
```

Comment:
Let us do not set the default value for `columnPruning`? We might lose the pruning opportunity if we call this constructor.

Reply:
The constructor with disabled `columnPruning` is called in the CSV writer and 30 times from test suites like `UnivocityParserSuite` and `CSVInferSchemaSuite`.

> We might lose the pruning opportunity if we call this constructor.

ok. I will enable it by default.

Reply:
always enabling it is also not right. Can we remove the default?

Reply:
removed